src/pudl/extract/ferc714.py

```diff
@@ -1,72 +1,97 @@
+FERC714_FILES: OrderedDict[str, dict[str, str]] = OrderedDict(
+    {
+        "id_certification_ferc714": {
+            "name": "Part 1 Schedule 1 - Identification Certification.csv",
+            "encoding": "iso-8859-1",
+        },
+        "gen_plants_ba_ferc714": {
+            "name": "Part 2 Schedule 1 - Balancing Authority Generating Plants.csv",
+            "encoding": "iso-8859-1",
+        },
+        "demand_monthly_ba_ferc714": {
+            "name": "Part 2 Schedule 2 - Balancing Authority Monthly Demand.csv",
+            "encoding": "utf-8",
+        },
+        "net_energy_load_ba_ferc714": {
+            "name": "Part 2 Schedule 3 - Balancing Authority Net Energy for Load.csv",
+            "encoding": "utf-8",
+        },
+        "adjacency_ba_ferc714": {
+            "name": "Part 2 Schedule 4 - Adjacent Balancing Authorities.csv",
+            "encoding": "iso-8859-1",
+        },
+        "interchange_ba_ferc714": {
+            "name": "Part 2 Schedule 5 - Balancing Authority Interchange.csv",
+            "encoding": "iso-8859-1",
+        },
+        "lambda_hourly_ba_ferc714": {
+            "name": "Part 2 Schedule 6 - Balancing Authority Hourly System Lambda.csv",
+            "encoding": "utf-8",
+        },
+        "lambda_description_ferc714": {
+            "name": "Part 2 Schedule 6 - System Lambda Description.csv",
+            "encoding": "iso-8859-1",
+        },
+        "description_pa_ferc714": {
+            "name": "Part 3 Schedule 1 - Planning Area Description.csv",
+            "encoding": "iso-8859-1",
+        },
+        "demand_forecast_pa_ferc714": {
+            "name": "Part 3 Schedule 2 - Planning Area Forecast Demand.csv",
+            "encoding": "utf-8",
+        },
+        "demand_hourly_pa_ferc714": {
+            "name": "Part 3 Schedule 2 - Planning Area Hourly Demand.csv",
+            "encoding": "utf-8",
+        },
+        "respondent_id_ferc714": {
+            "name": "Respondent IDs.csv",
+            "encoding": "utf-8",
+        },
+    }
+)
+"""Dictionary mapping PUDL tables to FERC-714 filenames and character encodings."""
 
-TABLE_ENCODING = {
-    "id_certification_ferc714": "iso-8859-1",
-    "gen_plants_ba_ferc714": "iso-8859-1",
-    "demand_monthly_ba_ferc714": None,
-    "net_energy_load_ba_ferc714": None,
-    "adjacency_ba_ferc714": "iso-8859-1",
-    "interchange_ba_ferc714": "iso-8859-1",
-    "lambda_hourly_ba_ferc714": None,
-    "lambda_description_ferc714": "iso-8859-1",
-    "description_pa_ferc714": "iso-8859-1",
-    "demand_forecast_pa_ferc714": None,
-    "demand_hourly_pa_ferc714": None,
-    "respondent_id_ferc714": None,
-}
-"""Dictionary describing the character encodings of the FERC 714 CSV files."""
 
-
-def extract(
-    ferc714_settings: Ferc714Settings = Ferc714Settings(), pudl_settings=None, ds=None
-):
+@multi_asset(
+    outs={"raw_" + table_name: AssetOut() for table_name in FERC714_FILES},
```

Comment:
Ehhhh, looking at this a bit more and comparing with the `eia860` example:

`eia860_raw_dfs` is a normal dictionary, which is being reconstructed out of a sorted list of dictionary items, but it's still just a plain vanilla dictionary when it's being iterated over in the `return` statement, which means that its ordering isn't guaranteed.

Similarly `outs` is just a normal dictionary with no guaranteed ordering -- in both modules.

But in the `ferc714` module, `FERC714_FILES` and `raw_dfs` are both `OrderedDicts` so they (and thus the tuple that is returned) is guaranteed to have the same orderings, while I think `outs` in the decorator is not (even though it's constructed from the ordered `FERC714_FILES`).

```python
    # create descriptive table_names
    eia860_raw_dfs = {
        "raw_" + table_name + "_eia860": df for table_name, df in eia860_raw_dfs.items()
    }
    eia860_raw_dfs = dict(sorted(eia860_raw_dfs.items()))

    return (
        Output(output_name=table_name, value=df)
        for table_name, df in eia860_raw_dfs.items()
    )
```

The fact that `outs` has keys that match the `output_name` in the returned `Output` values makes me wonder if this ordering is actually necessary. It seems like dagster has all the information it needs without the ordering, the names seem much more robust, and without the names the fact that `outs` is just a normal dictionary with no guaranteed order makes it seem like the ordering isn't guaranteed to work.

I can't find any mention of it in the [Output API docs](https://docs.dagster.io/_apidocs/ops#dagster.Output) or [Output Objects concept](https://docs.dagster.io/concepts/ops-jobs-graphs/op-events#output-objects) or the [multi_asset API docs](https://docs.dagster.io/_apidocs/assets#dagster.multi_asset).

Am I being crazy? Where'd you learn about it?

Reply:
I tried reversing the order of the outputs and it did complain (though the fact that it knows the name of both the expected result, and the one it got, and doesn't do the right thing seems odd).

I still think that `outs` and the tuple of `Output`s are guaranteed to have the same ordering here because they're `OrderedDict`s. Does that seem wrong to you?

Reply:
I feel like the [`multi_asset` guide](https://docs.dagster.io/concepts/assets/multi-assets) use to have a note about outs and Outputs needing to be in the same order. I might have learned about it from the slack or just running into an error.

Looks like they way you're sorting things is working 👍 I'll create a separate PR to updated how we're sorting the other multi_asset output keys.